### PR TITLE
[TST]: Wrong Corr with Timedelta index

### PR DESCRIPTION
**Pull request by phofl**
- [x] xref #31286 (closes the issue, if there is no way around the numerical issues)
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by phofl** [comment]
When using var instead of std, we can solve the numerical issues for the cases where var is still within floating point precision while std would not.

**Comment by jreback** [comment]
ok i'll buy this, why don't you add a bug fix note (in combination with the warning should be sufficient)

File: pandas/core/window/rolling.py [hunk 1]
```diff
@@ -1901,7 +1901,7 @@ def _get_corr(a, b):
                 window=window, min_periods=self.min_periods, center=self.center
             )
 
-            return a.cov(b, **kwargs) / (a.std(**kwargs) * b.std(**kwargs))
+            return a.cov(b, **kwargs) / (a.var(**kwargs) * b.var(**kwargs)) ** 0.5
```
Comment (jreback) [inline pandas/core/window/rolling.py:1904]: can you add some comments here (and ref this issue)

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/window/rolling.py (hunk 1 above)
Comment (phofl) [reply pandas/core/window/rolling.py:1906]: Done

**Comment by phofl** [comment]
Thanks, should we close this issue then?

**Comment by mroeschke** [comment]
Looks like this is still worthwhile. Can you merge in master and fix the merge conflicts?

**Comment by phofl** [comment]
cc @mroeschke Conflicts resolved

**Comment by mroeschke** [comment]
Sorry could you merge master one more time just to be sure. The travis failure doesn't look related to this PR

**Comment by phofl** [comment]
cc @mroeschke 

Merged master, checks failure unlrelated. Opened #37404 to clean it up

**Comment by phofl** [comment]
cc @mroeschke Travis successful. Merged master to show ci/checks is green now.

**Comment by mroeschke** [comment]
Looks like there's one more merge conflict

**Comment by phofl** [comment]
Done

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @phofl

**Pull request merged**